pkg/diff/diff.go

```diff
@@ -138,23 +140,26 @@ func WalkChan(dir string, ignores []string) <-chan *Entry {
 
 			var hash []byte
 
-			if info.Mode()&os.ModeSymlink == os.ModeSymlink {
-				hash, err = hashLink(path)
-			} else {
-				hash, err = hashFile(path)
-			}
-			if errors.Is(err, fs.ErrNotExist) {
-				return nil
-			}
-			if err != nil {
-				return pushErr(fmt.Errorf("hash file: %w", err))
+			if info.ModTime().UnixNano() >= latestModTime {
```

Comment:
Do we need to run a fancy filesystem in production to get nanonsecond precision in the mtime?

Reply:
> Since kernel 2.5.48, the stat structure supports nanosecond resolution for the three file timestamp fields. 

This seems to depend more on the kernel than the filesystem implementation, since it's calling out to a kernel syscall.

And it will also depend on the language / runtime, for example in Python the base `st_mtime` is second precision and they needed to add `st_mtime_nano` for nanosecond support.